stateoftheuniverse/widgets/exoplanet_count/exoplanet_count.py

```diff
@@ -60,62 +64,118 @@ def get_exoplanet_class(mass: u.Quantity,
+
+    def get_data(self,
+                 by_method: bool = True,
+                 by_class: bool = True):
+        """
+        Get the current total number of confirmed exoplanets from the
+        NASA Exoplanet Archive. Additionally, also get the number of
+        exoplanets grouped by detection method or by planet class (if
+        desired).
+
+        Args:
+            by_method: Whether or not to also return the exoplanet count
+                grouped by detection method.
+            by_class: Whether or not to also return the exoplanet count
+                grouped by (approximate) planet class.
+
+        Returns:
+            A dictionary with keys `{"total", "by_method", "by_class"}`
+            which holds the respective count of exoplanets.
+        """
+
+        # Instantiate the dictionary that will hold the results
+        exoplanet_count = dict()
+
+        # Get the confirmed exoplanets form the NASA Exoplanet Archive
+        confirmed_exoplanets = \
+            NasaExoplanetArchive.get_confirmed_planets_table()
+    
+        # Get the total count of all confirmed planets
+        exoplanet_count['total'] = len(confirmed_exoplanets)
+    
+        # Get count by detection method
+        if by_method:
+            exoplanet_count['by_method'] = \
+                dict(Counter(list(confirmed_exoplanets['pl_discmethod'])))
+    
+        # Get count by planet class (by looping over all confirmed exoplanets
+        # and classifying each of them based on its respective mass and radius)
+        if by_class:
+        
+            # Initialize the sub-dictionary that will hold the counts per class
+            exoplanet_count['by_class'] = dict()
+        
+            # Loop over all confirmed exoplanets and classify them individually
+            for i in range(len(confirmed_exoplanets)):
+            
+                # Get the mass and radius and cast to a astropy.units.Quantity
+                # Note: The NASA Exoplanet Archive usually returns values in
+                #       Jupiter units.
+                mass = float(
+                    str(confirmed_exoplanets[i]['pl_bmassj']).split()[0])
+                mass = u.Quantity(mass, u.jupiterMass)
+                radius = float(
+                    str(confirmed_exoplanets[i]['pl_radj']).split()[0])
+                radius = u.Quantity(radius, u.jupiterRad)
+            
+                # Classify the exoplanet based on these values
+                planet_class = get_exoplanet_class(mass=mass, radius=radius)
+            
+                # Increase the count for the planet class
+                if planet_class in exoplanet_count['by_class'].keys():
+                    exoplanet_count['by_class'][planet_class] += 1
+                else:
+                    exoplanet_count['by_class'][planet_class] = 1
+
+        # Store away the data retrieved by this method
+        self.data = exoplanet_count
+
+    def get_string(self):
```

Comment:
This entire class is dark magic to me, but I've seen the output and it looks good so I trust it (y)